Source/JavaScriptCore/runtime/LiteralParser.h

```diff
@@ -151,25 +151,34 @@ class LiteralParser {
     
     JSValue tryLiteralParse()
     {
+        ASSERT(m_mode == StrictJSON);
+        ASSERT(reviverMode == JSONReviverMode::Disabled);
         m_lexer.next();
-        JSValue result;
         VM& vm = getVM(m_globalObject);
-        if (m_mode == StrictJSON && reviverMode == JSONReviverMode::Disabled)
-            result = parseRecursivelyEntry(vm);
-        else {
-            result = parse(vm, StartParseStatement, nullptr);
-            if (m_lexer.currentToken()->type == TokSemi)
-                m_lexer.next();
-        }
+        JSValue result = parseRecursivelyEntry(vm);
         if (m_lexer.currentToken()->type != TokEnd)
             return JSValue();
         return result;
     }
 
     JSValue tryLiteralParse(JSONRanges* sourceRanges)
     {
+        ASSERT(m_mode == StrictJSON);
+        ASSERT(reviverMode == JSONReviverMode::Enabled);
         m_lexer.next();
-        JSValue result = parse(getVM(m_globalObject), m_mode == StrictJSON ? StartParseExpression : StartParseStatement, sourceRanges);
+        JSValue result = parse(getVM(m_globalObject), StartParseExpression, sourceRanges);
+        if (m_lexer.currentToken()->type != TokEnd)
+            return JSValue();
+        return result;
+    }
+
+    JSValue tryEval()
+    {
+        ASSERT(m_mode == SloppyJSON);
+        ASSERT(reviverMode == JSONReviverMode::Disabled);
```

Comment:
Ditto.  Make this a `static_assert` on `reviverMode`.

Reply:
Ditto.